addons/l10n_it_edi/tests/test_res_partner.py

```diff
@@ -79,3 +85,88 @@ def test_partner_l10n_it_codice_fiscale(self):
 
         partner_form.vat = 'IT12345670017'
         self.assertEqual(partner_form.l10n_it_codice_fiscale, '12345670017', "There was a typo in the VAT, changing it should change l10n_it_codice_fiscale as well")
+
+    def _test_normalized_data(self, testdata):
+        partner = self.env['res.partner'].create({'name': 'partner'})
+        for testentry in testdata:
+            with self.subTest(testentry=testentry):
+                partner.write({
+                    k: (v if isinstance(v, str | int | float) else v.id)
+                    for k, v in testentry.items() if not k.startswith('normalized')
+                })
+                for field, actual in partner._l10n_it_edi_get_values().items():
+                    if (expected := testentry.get(f'normalized_{field}')) is not None:
+                        self.assertEqual(expected, actual)
```

Comment:
`_l10n_it_edi_get_values()` can return quite a huge dictionnary and you will go through all the items of that dictionnary for each of your test entries just to test two or three values... I would rather do it the other way around, get the values of the `normalized_{field}` in your testentry and get those in the edi values dict. Small optimisation but the test are already taking so long, I think we can all benefit from this. Also you avoid the `is not None` for values that you don't want to test which was not great imo.

```suggestion
                l10n_it_edi_values = partner._l10n_it_edi_get_values()
                for field, expected in {k[11:]: v for k, v in testentry.items() if k.startswith("normalized_")}
                    self.assertEqual(expected, l10n_it_edi_values.get(field)
```